app/partials/app.cjsx

```diff
@@ -53,27 +53,25 @@ PanoptesApp = createReactClass
     auth.listen 'change', @handleAuthChange
     @handleAuthChange()
     generateSessionID()
+  
+  componentDidUpdate: ->
```

Comment:
This might be a bad idea.

Reply:
Did you test how often this callback is run when using the system? I'm wondering especially during classifying and how that might increase the talk api traffic? 

TBH getting notifications updates more often seems really useful, i'm just wondering about API calls and their frequency...

Reply:
Just noting that this code is currently in `componentWillReceiveProps`, which fires more often than `componentDidUpdate`. [`componentWillReceiveProps` is being removed in React 17](https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops), so this looks good to me.

Reply:
On second thoughts, why aren't these notifications using the sugar client?

Reply:
If this is the correct staging branch https://pr-5294.pfe-preview.zooniverse.org/ I think this line has caused an infinite loop. Check out the networks tab. I'm guessing the call to `notificationsCounter.update(user)` causes the component to update and make the call again. 

If that is the issue, you could add a condition to only make the call when the route changes, or something similar.

Reply:
Good point. `componentDidUpdate` will run after changes are batched and executed, including calls to `setState`. Any code inside `componentDidUpdate` should be guarded to avoid running on every update to state or props.

My preferred solution would be to use the sugar client to listen for notifications. The code is pretty much set up in the redux directory.

Second best solution would be to listen for notification changes on a time. Maybe [setInterval](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval) to update the counter on a fixed interval.

Reply:
I'd like to try using the sugar client, that just makes sense. I'll make some time, but until then I'm not sure what to do with this PR. I'm leaning towards closing for now, then will reopen once utilize the sugar client or setInterval. I'll prob do that (close for now) towards the end of today, unless I hear otherwise. 

A big thanks for comments though! We're a little closer to some much requested functionality.